java-util/src/main/java/io/druid/java/util/common/parsers/JSONPathParser.java

```diff
@@ -126,70 +127,85 @@ public void setFieldNames(Iterable<String> fieldNames)
     }
   }
 
-  private Map<String, Pair<FieldType, JsonPath>> generateFieldPaths(List<FieldSpec> fieldSpecs)
+  private Map<String, Pair<FieldType, FlattenExpr>> generateFieldPaths(List<FieldSpec> fieldSpecs)
   {
-    Map<String, Pair<FieldType, JsonPath>> map = new LinkedHashMap<>();
+    Map<String, Pair<FieldType, FlattenExpr>> map = new LinkedHashMap<>();
     for (FieldSpec fieldSpec : fieldSpecs) {
       String fieldName = fieldSpec.getName();
       if (map.get(fieldName) != null) {
         throw new IllegalArgumentException("Cannot have duplicate field definition: " + fieldName);
       }
-      JsonPath path = fieldSpec.getType() == FieldType.PATH ? JsonPath.compile(fieldSpec.getExpr()) : null;
-      Pair<FieldType, JsonPath> pair = new Pair<>(fieldSpec.getType(), path);
+      FlattenExpr path = null;
+      if (fieldSpec.getType() == FieldType.PATH) {
+        path = new FlattenExpr(JsonPath.compile(fieldSpec.getExpr()));
+      } else if (fieldSpec.getType() == FieldType.JQ) {
+        try {
+          path = new FlattenExpr(JsonQuery.compile(fieldSpec.getExpr()));
+        }
+        catch (JsonQueryException e) {
+          throw new ParseException(e, "Unable to compile JQ expression [%s]", fieldSpec.getExpr());
```

Comment:
It would be better for this to be an IllegalArgumentException. ParseExceptions are meant to represent parse errors on individual lines, but this error applies to the entire flatten spec.